### PR TITLE
Create Dockerfile

### LINKED ISSUE #17: Potential Improvement for `environment.yml` and `Dockerfile`
Thanks for setting up the environment @bfrizzell01. Just spotted 2 small issues:

1. `deepchecks` version confirmation in `environment.yml`
2. Automate the installation for `deepchecks` in `Dockerfile`

Please see comments on `environment.yml` and `Dockerfile` for detail.

### PR DESCRIPTION
**Pull request by bfrizzell01**
* added Python 3.11 dependency to `environment.yml`
* removed `conda-lock.yml`, replaced with `conda-linux-64.lock`
* added `Dockerfile`
* added docker publish action

### CODE REVIEW

**Comment by bfrizzell01** [comment]
Todo: Add `deepcheck` package to conda-lock

**Comment by bfrizzell01** [comment]
Also add brief instructions for running docker container

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,7 @@
+FROM quay.io/jupyter/minimal-notebook:afe30f0c9ad8
+
+COPY conda-linux-64.lock /tmp/conda-linux-64.lock
+RUN mamba update --quiet --file /tmp/conda-linux-64.lock \
+    && mamba clean --all -y -f \
+    && fix-permissions "${CONDA_DIR}" \
+    && fix-permissions "/home/${NB_USER}"
```
Comment (MasonZhang-MZ) [inline Dockerfile:7]: Might need to automate the installation for `deepchecks` during the build process by adding `RUN pip install deepchecks==0.3.2` at the end of the `Dockerfile` (for `deepchecks` version, please refer to the comment on the `environment.yml` file to confirm which version to install).

File: environment.yml [hunk 2]
```diff
@@ -6,6 +6,7 @@ dependencies:
...
   - scikit-learn=1.5.2
   - ucimlrepo=0.0.7
-  - ipykernel=6.29.5
-  - numpy=2.1.3
-  - scipy=1.14.1
\ No newline at end of file
+  - scipy=1.14.1
+  - python=3.11
+  - pandera=0.21.0
+  - deepchecks=0.3.2
```
Comment (MasonZhang-MZ) [inline environment.yml:12]: Just double-checking for the pinned version of `deepchecks`, any particular reason of using `0.3.2`? `0.18.1` should be the latest release, if there are any compatibility issue with this version, please disregard this comment.

**Review by MasonZhang-MZ — CHANGES_REQUESTED** [review]
Thanks for setting up the environment, Ben. Just spotted 2 small issues:

1. `deepchecks` version confirmation in `environment.yml`
2. Automate the installation for `deepchecks` in `Dockerfile`

Please see comments on `environment.yml` and `Dockerfile` for detail.

File: Dockerfile (hunk 1 above)
Comment (bfrizzell01) [reply Dockerfile:7]: Is the `deepchecks` package not included on your end when you run the container? It seems to properly install on my end. Let me know, and if you have issues I will make sure to implement this.

Update: refer to my following comment about the `environment.yml` file regarding this.

File: environment.yml (hunk 2 above)
Comment (bfrizzell01) [reply environment.yml:12]: `0.3.2` was the version that conda automatically installed into our environment with the other packages, so i'm assuming that is the version that is most compatible. But I can try it with the newer version and see if there are any issues.

Update: It does seem that version `0.18.1` has conflicts with other pinned package versions, so I've added a line to `Dockerfile` to include the newest version.

File: Dockerfile (hunk 1 above)
Comment (MasonZhang-MZ) [reply Dockerfile:7]: container runs fine on my end with `deepchecks` installed, thanks @bfrizzell01

Comment (bfrizzell01) [reply Dockerfile:7]: I've added the requested line to `Dockerfile`. Data Validation should now work properrly, @y1chi-z .

**Pull request merged**